com.unity.render-pipelines.universal/ShaderLibrary/Lighting.hlsl

```diff
@@ -627,21 +627,165 @@ half3 SampleLightmap(float2 lightmapUV, half3 normalWS)
+    if (weightProbe1 > 0.01)
+    {
+#ifdef _REFLECTION_PROBE_BOX_PROJECTION
+        reflectVector = BoxProjectedCubemapDirection(originalReflectVector, positionWS, unity_SpecCube1_ProbePosition, unity_SpecCube1_BoxMin, unity_SpecCube1_BoxMax);
+#else
+        reflectVector = originalReflectVector;
+#endif // _REFLECTION_PROBE_BOX_PROJECTION
+        half4 encodedIrradiance = half4(SAMPLE_TEXTURECUBE_LOD(unity_SpecCube1, samplerunity_SpecCube1, reflectVector, mip));
+
+#if defined(UNITY_USE_NATIVE_HDR) || defined(UNITY_DOTS_INSTANCING_ENABLED)
+        irradiance += weightProbe1 * encodedIrradiance.rbg;
+#else
+        irradiance += weightProbe1 * DecodeHDREnvironment(encodedIrradiance, unity_SpecCube1_HDR);
+#endif // UNITY_USE_NATIVE_HDR || UNITY_DOTS_INSTANCING_ENABLED
+    }
+
+    // Use any remaining weight to blend to environment reflection cube map
+    if (totalWeight < 0.99)
+    {
+        reflectVector = originalReflectVector;
+        half4 encodedIrradiance = half4(SAMPLE_TEXTURECUBE_LOD(_GlossyEnvironmentCubeMap, sampler_GlossyEnvironmentCubeMap, reflectVector, mip));
+
+#if defined(UNITY_USE_NATIVE_HDR) || defined(UNITY_DOTS_INSTANCING_ENABLED)
+        irradiance += (1 - totalWeight) * encodedIrradiance.rbg;
+#else
+        irradiance += (1 - totalWeight) * DecodeHDREnvironment(encodedIrradiance, _GlossyEnvironmentCubeMap_HDR);
+#endif // UNITY_USE_NATIVE_HDR || UNITY_DOTS_INSTANCING_ENABLED
+    }
+
+    return irradiance;
+}
+
+half3 GlossyEnvironmentReflection(half3 reflectVector, float3 positionWS, half perceptualRoughness, half occlusion)
+{
+#if !defined(_ENVIRONMENTREFLECTIONS_OFF)
+    half3 irradiance = half3(0, 0, 0);
+
+#ifdef _REFLECTION_PROBE_BLENDING
+    irradiance = CalculateIrradianceFromReflectionProbes(reflectVector, positionWS, perceptualRoughness);
+#else
+#ifdef _REFLECTION_PROBE_BOX_PROJECTION
+    reflectVector = BoxProjectedCubemapDirection(reflectVector, positionWS, unity_SpecCube0_ProbePosition, unity_SpecCube0_BoxMin, unity_SpecCube0_BoxMax);
+#endif // _REFLECTION_PROBE_BOX_PROJECTION
+    half mip = PerceptualRoughnessToMipmapLevel(perceptualRoughness);
+    half4 encodedIrradiance = half4(SAMPLE_TEXTURECUBE_LOD(unity_SpecCube0, samplerunity_SpecCube0, reflectVector, mip));
+
+    //TODO:DOTS - we need to port probes to live in c# so we can manage this manually.
+#if defined(UNITY_USE_NATIVE_HDR) || defined(UNITY_DOTS_INSTANCING_ENABLED)
+    irradiance = encodedIrradiance.rgb;
+#else
+    irradiance = DecodeHDREnvironment(encodedIrradiance, unity_SpecCube0_HDR);
+#endif // UNITY_USE_NATIVE_HDR || UNITY_DOTS_INSTANCING_ENABLED
+#endif // _REFLECTION_PROBE_BLENDING
+
+    return irradiance * occlusion;
+#endif // _ENVIRONMENTREFLECTIONS_OFF
+
+    return _GlossyEnvironmentColor.rgb * occlusion;
+}
 half3 GlossyEnvironmentReflection(half3 reflectVector, half perceptualRoughness, half occlusion)
 {
 #if !defined(_ENVIRONMENTREFLECTIONS_OFF)
+    half3 irradiance = half3(0, 0, 0);
```

Comment:
Is the zero assignment here necessary? I think you can just declare it since we always assign below. These type of redundant assignments/memsets can build up.

Reply:
resolved in 203240f